ic_routing_board_generation/board_generator/board_generator_v3_0_0_rb.py

```diff
@@ -0,0 +1,476 @@
+        """ Returns the number of cells adjacent to the input cell which are in the wire_list.
+
+        Args:
+            input (Position): The input cell to search adjacent to.
+            wire_list (List): List of cells already in the wire.
+
+        Returns:
+            int: Number of adjacent cells that are in the wire_list.
+        """
+        num_adjacent = 0
+        # Check above, below, to the left and the right and count the number in the wire_list.
+        if (input.x-1, input.y) in wire_list:
+            num_adjacent += 1
+        if (input.x+1, input.y) in wire_list:
+            num_adjacent += 1
+        if (input.x, input.y-1) in wire_list:
+            num_adjacent += 1
+        if (input.x, input.y+1) in wire_list:
+            num_adjacent += 1
+        return num_adjacent
+
+    # The next six methods support the add_wire_head_target_erode method.
+    # Currently, that method leaves extraneous loops, so it's in-progress.
+    # Also, the recursive limit on the connectible_cells errors out after 1000 recursions.
+    def connectible_cells(self, x_head: int, y_head: int) -> List:
+        """ Return a list of 2D tuples, cells that are connectible to (x_head, y_head).
+
+        Args:
+            x_head, y_head (int, int) : cell to connect to.
+
+        Returns:
+            List[(int,int)...] : output list of connected cells.
+        """
+        connectible_list = []
+        self.add_connectible_cell(x_head, y_head, connectible_list)
+        return connectible_list
+
+    def add_connectible_cell(self, x_pos: int, y_pos: int, connectible_list: List) -> List:
+        """ Add the specified cell to the list, recursively call adjacent cells, and return list.
+
+        Args:
+            x_pos, y_pos (int, int) : cell to add to the list.
+            connectible_list (List[(int,int)...] : input list of connected cells.
+
+        Returns:
+            List[(int,int)...] : output list of connected cells.
+        """
+        if (x_pos, y_pos) in connectible_list:
+            return connectible_list
+        connectible_list.append((x_pos, y_pos))
+        # Recursively add the cells above, to the right, below, and to the left if they're valid and open
+        if (x_pos < self.dim.x-1) and (self.layout[x_pos+1, y_pos] == EMPTY) and (x_pos+1, y_pos) not in connectible_list:
+            self.add_connectible_cell(x_pos+1, y_pos, connectible_list)
+        if (y_pos < self.dim.y-1) and (self.layout[x_pos, y_pos+1] == EMPTY) and (x_pos, y_pos+1) not in connectible_list:
+            self.add_connectible_cell(x_pos, y_pos+1, connectible_list)
+        if (x_pos > 0) and (self.layout[x_pos-1, y_pos] == EMPTY) and (x_pos-1, y_pos) not in connectible_list:
+            self.add_connectible_cell(x_pos-1, y_pos, connectible_list)
+        if (y_pos > 0) and (self.layout[x_pos, y_pos-1] == EMPTY) and (x_pos, y_pos-1) not in connectible_list:
+            self.add_connectible_cell(x_pos, y_pos-1, connectible_list)
```

Comment:
feels like (x_pos < self.dim.x-1) and (self.layout[x_pos+1, y_pos] == EMPTY) and (x_pos+1, y_pos) not in connectible_list could be wrapped in a function and you pass the arguments that you want to evaluate

Reply:
Done